dts/bindings/mtd/atmel,at24-smbus.yaml

```diff
@@ -0,0 +1,9 @@
+# Copyright (c) 2023 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+description: Atmel AT24 - compatible SMBus EEPROM
+
+compatible: "atmel,at24"
+
+include: ["atmel,at2x-base.yaml", smbus-device.yaml]
```

Comment:
This is yet another annoyance with the way we ended up add SMbus support - now we need different bindings for exactly the same device on what is essentially the same type of bus, but with a different API.

I am still not sold on this at all, sorry.

Reply:
My concern is duplication efforts for I2C devices on a SMBus, which is basically just a restricted protocol on top of I2C. I am questioning if the way we decided to implement SMBus support in Zephyr, which leads to these duplication efforts, is the correct architecture.

Reply:
What is exactly "duplicated effort"? ;) Did I add 100% more code to the driver?

Reply:
Duplicated, as in you needed to modify the binding and driver at all. This is the exact same device on the same type of bus, but with restrictions. This shows that _each_ driver, which a user need to support on the SMBus, requires driver changes.

From a quick grep in the source tree, we have 160 I2C device drivers. In order to support these devices on an SMBus, each of these bindings and drivers would need to be modified.

Reply:
What exact `each driver` are we talking about here? So far there is no use case AFAIK. Even current at24 driver is for qemu for testing purposes only.
Real EEPROM driver on SMBus would need to switch pages using specific smbus commands like it is done in Linux:
https://github.com/torvalds/linux/blob/44c026a73be8038f03dbdeef028b642880cf1511/drivers/misc/eeprom/ee1004.c#L83

Reply:
Then I am even more puzzled about this PR. Why not add a "real" SMBus EEPROM driver, then?

Reply:
OK, I close PR since I used it only for testing with Qemu